crates/js-component-bindgen/src/function_bindgen.rs

```diff
@@ -1048,7 +1050,25 @@ impl Bindgen for FunctionBindgen<'_> {
             Instruction::CallWasm { sig, .. } => {
                 let sig_results_length = sig.results.len();
                 self.bind_results(sig_results_length, results);
-                uwriteln!(self.src, "{}({});", self.callee, operands.join(", "));
+                if self.is_async {
+                    if self.use_asyncify {
+                        uwriteln!(
+                            self.src,
+                            "await asyncifyWrapExport({})({});",
+                            self.callee,
+                            operands.join(", ")
+                        );
+                    } else {
+                        uwriteln!(
+                            self.src,
+                            "await WebAssembly.promising({})({});",
```

Comment:
Ideally this wrapper and the one above should be applied before we even get to the `callee` so that the WebAssembly.promising wrapper is cached at initialization time. Basically where the callee is first constructed.

It might be possible to just change top-level function definitions like the following:

```js
function binding () { .... }
```

to

```js
const binding = WebAssembly.promising(function binding () { ... });
```

Reply:
Fixed